pyaedt/modules/Mesh.py

```diff
@@ -26,18 +27,63 @@
     "2D Extractor": "MeshSetup",
 }
 
+mesh_props = {
+    "CurvedSurfaceApproxChoice": "Curved Mesh Approximation Type",
+    "SliderMeshSettings": "Curved Surface Mesh Resolution",
+    "SurfDevChoice": "Surface Deviation Choice",
+    "SurfDev": "Surface Deviation",
+    "NormalDevChoice": "Normal Deviation Choice",
+    "NormalDev": "Normal Deviation",
+    "AspectRatioChoice": "Aspect Ratio Choice",
+    "AspectRatio": "Aspect Ratio",
+    "UseAutoLength": "Use Auto Simplify",
+    "DefeatureLength": "Model Resolution Length",
+    "SurfaceRepPriority": "Surface Representation Priority for TAU",
+    "RestrictLength": "Restrict Length",
+    "MaxLength": "Max Length",
+    "RestrictElem": "Restrict Max Elems",
+    "SkinDepth": "Skin Depth",
+    "NumLayers": "Num Layers",
+    "SurfTriMaxLength": "Max Elem Length",
+    "NumMaxElem": "Max Elems",
+    "Apply": "Apply Curvilinear Elements",
+    "DisableForFacetedSurfaces": "Disable for Faceted Surface",
+    "Total Layer Thickness": "Total Layer Thickness",
+    "Number of Layers": "Number of Layers",
+    "RestrictMaxElemLength": "Restrict Max Element Length",
+    "MaxElemLength": "Max Element Length",
+    "RestrictLayersNum": "Restrict Layers Number",
+    "LayersNum": "Number of layers",
+    "Layer Thickness": "Layer Thickness",
+}
+
 
 class MeshProps(OrderedDict):
     """AEDT Mesh Component Internal Parameters."""
 
     def __setitem__(self, key, value):
+        pyaedt_mesh_attr = list(self._pyaedt_mesh.__dict__.keys())
         OrderedDict.__setitem__(self, key, value)
         if key in ["Edges", "Faces", "Objects"]:
-            res = self._pyaedt_mesh.update_assignment()
+            if "_meshicepak" in pyaedt_mesh_attr or "_mesh3dlayout" in pyaedt_mesh_attr:
+                self._pyaedt_mesh.update_assignment()
+            else:
+                self._pyaedt_mesh._mesh.omeshmodule.ReassignOp(self._pyaedt_mesh.name, ["{}:=".format(key), value])
         else:
-            res = self._pyaedt_mesh.update()
-        if not res:
-            self._pyaedt_mesh._app.logger.warning("Update of %s Failed. Check needed arguments", key)
+            if "_meshicepak" in pyaedt_mesh_attr or "_mesh3dlayout" in pyaedt_mesh_attr:
+                self._pyaedt_mesh.update()
+            else:
+                mesh_obj = self._pyaedt_mesh._mesh._app.odesign.GetChildObject("Mesh").GetChildObject(
+                    self._pyaedt_mesh.name
+                )
+                if key in mesh_props.keys():
+                    if key == "SurfaceRepPriority":
+                        if value == 0:
+                            value = "Normal"
+                        else:
+                            value = "High"
+
+                    mesh_obj.SetPropValue(mesh_props[key], value)
```

Comment:
Not sure if this has to be inside a try, except because there are some properties sometimes, readonly and the SetPropValue fails